fixtures/nelson.py

```diff
@@ -7,12 +9,39 @@
 from sphinxcontrib.napoleon import docstring
 from sphinxcontrib.napoleon.docstring import NumpyDocstring
 import sphinx
+import yaml
 
 from utils.log import get_rel_path, logger
 
 config = Config(napoleon_use_param=True, napoleon_use_rtype=True)
 
 
+def get_meta(obj):
+    doc = getattr(obj, '__doc__') or ''
+    p = GoogleDocstring(stripper(doc), config)
+    return p.metadata
+
+
+def pytest_collection_modifyitems(items):
+    output = {}
+    for item in items:
+        item_class = item.location[0]
+        item_class = item_class[:item_class.rfind('.')].replace('/', '.')
+        item_name = item.location[2]
+        item_param = re.findall('\.*(\[.*\])', item_name)
+        if item_param:
+            item_name = item_name.replace(item_param[0], '')
+        node_name = '{}.{}'.format(item_class, item_name)
+        output[node_name] = {}
+        output[node_name]['docstring'] = base64.b64encode(getattr(item.function, '__doc__') or '')
+        output[node_name]['name'] = item_name
+        doc_meta = {k: v for k, v in item._metadata.get('from_docs', {}).items()}
+        output[node_name]['metadata'] = {'from_docs': doc_meta}
+
+    with open('doc_data.yaml', 'w') as f:
+        yaml.dump(output, f)
```

Comment:
You might be able to let AttrDict do the work here, and maybe this will help with your serialization problems above. I doubt it, but it could be worth a shot.

`attrdict_instance.dump(open('doc_data.yaml', 'w'))`